### PR TITLE
Allow column config in serial syntax

### PR DESCRIPTION
**Pull request by annda**
The same columns are used in aggregation and new entry editor

### CODE REVIEW

File: script/AggregationEditor.js [hunk 1]
```diff
@@ -140,7 +142,7 @@ const AggregationEditor = function (idx, table) {
         DOKU_BASE + 'lib/exe/ajax.php',
         {
             call: 'plugin_struct_aggregationeditor_new',
-            schema: schema
+            data: {schema: schema, searchconf: searchconf}
```
Comment (splitbrain) [inline script/AggregationEditor.js:145]: Isn't the schema part of the SearchConfig? Wouldn't it be enough to pass the latter param?

Comment (annda) [reply script/AggregationEditor.js:145]: SearchConf contains an array of schemas and I'd rather not refactor the code at this point.

File: action/aggregationeditor.php [hunk 2]
```diff
@@ -164,7 +164,18 @@ protected function newRowEditor()
         echo '<legend>' . $this->getLang('lookup new entry') . '</legend>';
         /** @var action_plugin_struct_edit $edit */
         $edit = plugin_load('action', 'struct_edit');
-        foreach ($schema->getColumns(false) as $column) {
+
+        // filter columns based on searchconf cols from syntax
+        $columns = [];
+        if (!empty($INPUT->arr('data')['searchconf']['cols']) && is_array($INPUT->arr('data')['searchconf']['cols'])) {
+            foreach ($INPUT->arr('data')['searchconf']['cols'] as $col) {
+                $columns[] = $schema->findColumn($col);
```
[the remark this answers is not in the document]
Comment (annda) [reply action/aggregationeditor.php]: I added a filter for empty/false columns.

File: script/AggregationEditor.js (hunk 1 above)
Comment (splitbrain) [reply script/AggregationEditor.js:145]: Hmm. We only support a single schema anyway, right? Can't you simply access searchconf->schemas[0]? Maybe I misunderstand.

**Pull request merged**